designer/client/conditions/SelectConditions.tsx

```diff
@@ -82,25 +88,22 @@ class SelectConditions extends React.Component<Props, State> {
     const fields: any = Object.values(this.fieldsForPath(path));
     const { conditions = [] } = data;
     let conditionsForPath: any[] = [];
-    const stringConditions = conditions.filter(
-      (condition) => typeof condition.value === "string"
-    );
-    // nested conditions have their own data structure, so need to be considered separately
-    const nestedConditions = conditions.filter(
-      (condition) =>
-        typeof condition.value !== "string" &&
-        condition.value.conditions
-          .map((innerCondition) =>
-            innerCondition.hasOwnProperty("conditionName")
-          )
-          .includes(true)
-    );
-    const objectConditions = conditions.filter(
-      (condition) =>
-        typeof condition.value !== "string" &&
-        !nestedConditions.find(
-          (nestedCondition) => nestedCondition.name === condition.name
-        )
+    const [
+      stringConditions,
+      nestedConditions,
+      objectConditions,
+    ] = conditions.reduce(
+      (res, condition) => {
+        if (condition.value === "string") {
+          res[0].push(condition);
+        } else if (!!hasNestedCondition(condition)) {
+          res[1].push(condition);
+        } else if (isObjectCondition(condition)) {
+          res[2].push(condition);
+        }
+        return res;
+      },
+      [[], [], []]
```

Comment:
I'm ok with reduce approach as well, but object will be nicer for readability, instead of a nested array. Doing res[0].push, means I have to scroll up and check what that refers to.

```.ts

const conditionsByTypeMap = conditions.reduce((conditionsByType, condition) => {
  
  if(isStringCondition(condition)) {
    conditionsByType.string.push(condition)
    return; // maybe return conditionsByType? double check that
  }

  if(hasNestedCondition(condition)) {
    conditionsByType.nested.push(condition)
    return;
  } 
  
  conditionsByType.object.push(condition)
  

}, {string: [], nested: [], object: []})
// FWIW - I purposefully do not name these keys whateverConditions, since you will end up with context duplication. e.g. conditionsByTypeMap.stringConditions. 
// I already know I am in the context of conditions (conditionsByTypeMap), so I don't need more words to tell me I'm looking at a condition (stringConditions). 

// but if you never use the object like that, then I'm also fine with
const { stringConditions, ... } = conditions.reduce((acc, curr) => {...}, { stringConditions: [], ...})

// or just use 
conditionsByTypeMap.string
conditionsByTypeMap.nested

```